Cortex XDR CCP content creation

ltipp147 opened the pull request:
Required items, please complete
   
   Change(s):
   - Added content structure for Sentinel solution - Cortex XDR (ONE Data Connector and THREE Analytic Rules).
   - Added Cortex XDR Logo in the folder \Azure-Sentinel\Logos
   - Added sample data in the folder Azure-Sentinel\Sample Data\Custom
   - Added custom logs table schema in the folder Azure-Sentinel\.script\tests\KqlvalidationsTests\CustomTables

   Reason for Change(s):
   - Initial commit

   Version Updated:
   - N/A

   Testing Completed:
   - Yes

   Checked that the validations are passing and have addressed any issues that are present:
   - Yes

ltipp147 commented:
>

ltipp147 commented:
@microsoft-github-policy-service agree company="DEFEND Ltd."

L Tippayaratprontawee | Cybersecurity Analyst | +64 211 412 122 | DEFEND

From: microsoft-github-policy-service[bot] ***@***.***>
Sent: Wednesday, June 28, 2023 11:15 AM
To: Azure/Azure-Sentinel ***@***.***>
Cc: L Tippayaratprontawee ***@***.***>; Mention ***@***.***>
Subject: Re: [Azure/Azure-Sentinel] Cortex XDR CCP content creation (PR #8373)

CYBERSECURITY WARNING
This email originated from outside our organisation. Please take extra care when clicking on any links or opening any attachments.

@ltipp147<https://github.com/ltipp147> please read the following Contributor License Agreement(CLA). If you agree with the CLA, please reply with the following information.

@microsoft-github-policy-service agree [company="{your company}"]

Options:

  *   (default - no company specified) I have sole ownership of intellectual property rights to my Submissions and I am not making Submissions in the course of work for my employer.

@microsoft-github-policy-service agree

  *   (when company given) I am making Submissions in the course of work for my employer (or my employer has intellectual property rights in my Submissions by contract or applicable law). I have permission from my employer to make Submissions and enter into this Agreement on behalf of my employer. By signing below, the defined term "You" includes me and my employer.

@microsoft-github-policy-service agree company="Microsoft"
Contributor License Agreement
Contribution License Agreement

This Contribution License Agreement ("Agreement") is agreed to by the party signing below ("You"),
and conveys certain license rights to Microsoft Corporation and its affiliates ("Microsoft") for Your
contributions to Microsoft open source projects. This Agreement is effective as of the latest signature
date below.

  1.  Definitions.
"Code" means the computer software code, whether in human-readable or machine-executable form,
that is delivered by You to Microsoft under this Agreement.
"Project" means any of the projects owned or managed by Microsoft and offered under a license
approved by the Open Source Initiative (www.opensource.org<http://www.opensource.org/>).
"Submit" is the act of uploading, submitting, transmitting, or distributing code or other content to any
Project, including but not limited to communication on electronic mailing lists, source code control
systems, and issue tracking systems that are managed by, or on behalf of, the Project for the purpose of
discussing and improving that Project, but excluding communication that is conspicuously marked or
otherwise designated in writing by You as "Not a Submission."
"Submission" means the Code and any other copyrightable material Submitted by You, including any
associated comments and documentation.
  2.  Your Submission. You must agree to the terms of this Agreement before making a Submission to any
Project. This Agreement covers any and all Submissions that You, now or in the future (except as
described in Section 4 below), Submit to any Project.
  3.  Originality of Work. You represent that each of Your Submissions is entirely Your original work.
Should You wish to Submit materials that are not Your original work, You may Submit them separately
to the Project if You (a) retain all copyright and license information that was in the materials as You
received them, (b) in the description accompanying Your Submission, include the phrase "Submission
containing materials of a third party:" followed by the names of the third party and any licenses or other
restrictions of which You are aware, and (c) follow any other instructions in the Project's written
guidelines concerning Submissions.
  4.  Your Employer. References to "employer" in this Agreement include Your employer or anyone else
for whom You are acting in making Your Submission, e.g. as a contractor, vendor, or agent. If Your
Submission is made in the course of Your work for an employer or Your employer has intellectual
property rights in Your Submission by contract or applicable law, You must secure permission from Your
employer to make the Submission before signing this Agreement. In that case, the term "You" in this
Agreement will refer to You and the employer collectively. If You change employers in the future and
desire to Submit additional Submissions for the new employer, then You agree to sign a new Agreement
and secure permission from the new employer before Submitting those Submissions.
  5.  Licenses.

  *   Copyright License. You grant Microsoft, and those who receive the Submission directly or
indirectly from Microsoft, a perpetual, worldwide, non-exclusive, royalty-free, irrevocable license in the
Submission to reproduce, prepare derivative works of, publicly display, publicly perform, and distribute
the Submission and such derivative works, and to sublicense any or all of the foregoing rights to third
parties.
  *   Patent License. You grant Microsoft, and those who receive the Submission directly or
indirectly from Microsoft, a perpetual, worldwide, non-exclusive, royalty-free, irrevocable license under
Your patent claims that are necessarily infringed by the Submission or the combination of the
Submission with the Project to which it was Submitted to make, have made, use, offer to sell, sell and
import or otherwise dispose of the Submission alone or with the Project.
  *   Other Rights Reserved. Each party reserves all rights not expressly granted in this Agreement.
No additional licenses or rights whatsoever (including, without limitation, any implied licenses) are
granted by implication, exhaustion, estoppel or otherwise.

  1.  Representations and Warranties. You represent that You are legally entitled to grant the above
licenses. You represent that each of Your Submissions is entirely Your original work (except as You may
have disclosed under Section 3). You represent that You have secured permission from Your employer to
make the Submission in cases where Your Submission is made in the course of Your work for Your
employer or Your employer has intellectual property rights in Your Submission by contract or applicable
law. If You are signing this Agreement on behalf of Your employer, You represent and warrant that You
have the necessary authority to bind the listed employer to the obligations contained in this Agreement.
You are not expected to provide support for Your Submission, unless You choose to do so. UNLESS
REQUIRED BY APPLICABLE LAW OR AGREED TO IN WRITING, AND EXCEPT FOR THE WARRANTIES
EXPRESSLY STATED IN SECTIONS 3, 4, AND 6, THE SUBMISSION PROVIDED UNDER THIS AGREEMENT IS
PROVIDED WITHOUT WARRANTY OF ANY KIND, INCLUDING, BUT NOT LIMITED TO, ANY WARRANTY OF
NONINFRINGEMENT, MERCHANTABILITY, OR FITNESS FOR A PARTICULAR PURPOSE.
  2.  Notice to Microsoft. You agree to notify Microsoft in writing of any facts or circumstances of which
You later become aware that would make Your representations in this Agreement inaccurate in any
respect.
  3.  Information about Submissions. You agree that contributions to Projects and information about
contributions may be maintained indefinitely and disclosed publicly, including Your name and other
information that You submit with Your Submission.
  4.  Governing Law/Jurisdiction. This Agreement is governed by the laws of the State of Washington, and
the parties consent to exclusive jurisdiction and venue in the federal courts sitting in King County,
Washington, unless no federal subject matter jurisdiction exists, in which case the parties consent to
exclusive jurisdiction and venue in the Superior Court of King County, Washington. The parties waive all
defenses of lack of personal jurisdiction and forum non-conveniens.
  5.  Entire Agreement/Assignment. This Agreement is the entire agreement between the parties, and
supersedes any and all prior agreements, understandings or communications, written or oral, between
the parties relating to the subject matter hereof. This Agreement may be assigned by Microsoft.

-
Reply to this email directly, view it on GitHub<https://github.com/Azure/Azure-Sentinel/pull/8373#issuecomment-1610344139>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/BA3UOEZ7UZL57JUNYFZPY53XNNSODANCNFSM6AAAAAAZU2LWIU>.
You are receiving this because you were mentioned.Message ID: ***@***.******@***.***>>

This email and any files transmitted with it are confidential and intended solely for the use of the individual or entity to whom they are addressed. If you have received this email in error please notify the system manager. This message contains confidential information and is intended only for the individual named. If you are not the named addressee you should not disseminate, distribute or copy this e-mail. Please notify the sender immediately by e-mail if you have received this e-mail by mistake and delete this e-mail from your system. If you are not the intended recipient you are notified that disclosing, copying, distributing or taking any action in reliance on the contents of this information is strictly prohibited.

v-prasadboke commented:
Hello @ltipp147 please repackage the solution. 
Please add solutionmetadata and Input file in the solution folder.
To repackage the solution please go through this Readme.md file
https://github.com/Azure/Azure-Sentinel/blob/master/Tools/Create-Azure-Sentinel-Solution/V2/README.md

You can refer to any of the solution folder for folder structure

v-prasadboke commented:
Hello @ltipp147 any updates on the above

ltipp147 commented:
Hi @v-prasadboke, apologies for late response - we're working on it and will add the package within this week, thanks!

ltipp147 commented:
Hi @v-prasadboke the solution has been repackaged with SolutionMetadata and Input file in the solution folder. Please, any feedback is appreciated.

v-prasadboke commented on Solutions/Cortex XDR/SolutionMetadata.json:
Hello @ltipp147 solution metadata should be in solution folder, not in package.
Package folder should contain maintemplate and createui and a zip package

v-prasadboke commented:
Please update the branch from master so that i can pull it my local

v-prasadboke commented:
Hello @ltipp147 can you please specify the sampledata in which analytic rule is being used

ltipp147 commented:
Hi @v-prasadboke, I believe I have satisfied the requested requirements. Please see the screenshot below for the folder structure and sample data that have already been uploaded:

- The SolutionMetadata.json is in the Solution (Cortex XDR) folder
- The Package folder contains (1) mainTemplate.json, (2) createUiDefinition.json, and (3) a zip package of version 3.0.0 (generated by createSolutionV3.ps1 toolkit)
- The sample data in which analytic rule is being used has been uploaded in the _Sample Data/Custom_ folder; these include (1) Raw Logs, (2) Ingested Logs, and (3) Schema.

<img width="153" alt="CortexXDRSolution" src="https://github.com/Azure/Azure-Sentinel/assets/137840403/12b529cf-a82f-44e2-8757-fd0f7b3d2be6">

v-prasadboke commented:
Thank you for the clarification @ltipp147, Will update you shortly on this

v-prasadboke commented:
Hello @ltipp147 it seems, this solution have a data connector and analytic rules are dependent on it, i request you to fill the data mentioned in all the analytic rules. 
Data to be filled :- requiredDataConnectors, data types, tactics: [] & relevantTechniques: [],

You can refer this Detection for more clarification 
https://github.com/Azure/Azure-Sentinel/blob/master/Solutions/Network%20Session%20Essentials/Analytic%20Rules/AnomalyFoundInNetworkSessionTraffic.yaml

ltipp147 commented:
Hi @v-prasadboke, I have updated the solution as requested:
- Added requiredDataConnectors and dataTypes to all analytic rules
- Added data connector ID 'CortexXDR' to the list in 'ValidConnectorIds.json' file to fix DetectionTemplateSchemaValidation fail

Let me know if more detail is required, thanks!

v-prasadboke commented:
Hello @ltipp147, Thank you for committing the changes.
Will be investigating this PR, you can expect an update till 25 Jul 2023

v-prasadboke commented:
Hello @ltipp147 Still data fields are missing for tactics: [] , relevantTechniques: [] from all the analytic rules

v-prasadboke commented on Solutions/Cortex XDR/Analytic Rules/CortexXDR_High.yaml:
```diff
@@ -0,0 +1,51 @@
+triggerThreshold: 0
+suppressionDuration: 5h
+suppressionEnabled: false
+tactics: []
```
This field is missing

ltipp147 replied:
Hi @v-prasadboke, tactics have been updated for all three analytic rules as requested

v-prasadboke commented:
Thank you @ltipp147 for committing the changes, you can expect an update till 26 Jul 2023

v-prasadboke commented:
Hello @ltipp147 you still havent filled relevant techniques of all the analytic rules.
Can you please add the required fields.

dark-binary commented:
Hey @v-prasadboke , are tactics and technique a mandatory field for the solution? The reason I ask is because this solution aims to pull incidents from Cortex XDR (and each incident might come up with different tactic and technique mapped with it). We can add a generic technique that MIGHT be suitable for all incidents, but would be on a high level rather than a targeted mapping. Would prefer leaving out the tactics and/or technique field empty if it isn't mandatory, or we are happy to add a generic mapping of tactic and technique if that will help pass the solution.

v-prasadboke commented:
Hello @dark-binary, I got the issue you are facing. Just give me some time, Will update about this within a short span of time.

v-prasadboke commented:
Hello @dark-binary, you can keep it empty or we have another option of dynamic mapping of techniques and tactics.

You can go through this link for customizing the alert.
https://learn.microsoft.com/en-us/azure/sentinel/customize-alert-details

dark-binary commented:
Hey @v-prasadboke , it would be best kept empty for the moment. We will work on dynamic mapping as an improvement, since the values from the alert cannot be directly applied for the fields (needs to be sanitized). 

Can we skip the tactics and technique mapping for now, and I can have our dev team work on the dynamic mapping from the alert as an improvement later, if that works?

v-prasadboke commented:
Ok sure @dark-binary, ill continue with the same and update you till EOD.

v-prasadboke commented:
Hello @dark-binary can you please add release notes in the PR.
Guidance to release notes - https://github.com/Azure/Azure-Sentinel/blob/master/Solutions/ReleaseNotesGuidance.md

And also update the zip package with latest createui and maintemplate. 
Createui and maintemplate in the zip should match with createui and maintemplate outside the zip

ltipp147 commented:
Hey @v-prasadboke the latest pull request has the below updates:

1 - Removed tactics from the 3 analytic rules (these will be added along with techniques as an improvement in the next version)
2 - Added ReleaseNotes.md
3 - Updated the zip package to reflect the latest changes in maintemplate and createui files

Hopefully everything is good. Please let us know if you have any questions.

v-prasadboke commented:
Thank you for updating @ltipp147, Will have a look at this and let you know till EOD.

ltipp147 commented:
Hi @v-prasadboke, apologize for the oversight in updating the files. The necessary updates have now been completed to reflect the most current information:

1 - Added the ReleaseNotes hyperlink to the CreateUiDefinition.json file
2 - Updated the zip package to match the mainTemplate and CreateUiDefinition files in and outside the zip folder

Please let us know if you have any further questions regarding the updated files. Thank you!

v-prasadboke reviewed (approved):
All looks good
